connectors/src/connectors/microsoft/temporal/spreadsheets.ts

```diff
@@ -0,0 +1,232 @@
+import { upsertTableFromCsv } from "@connectors/lib/data_sources";
+import logger from "@connectors/logger/logger";
+import { ConnectorResource } from "@connectors/resources/connector_resource";
+import type { MicrosoftRootResource } from "@connectors/resources/microsoft_resource";
+import { MicrosoftNodeResource } from "@connectors/resources/microsoft_resource";
+
+async function upsertSpreadsheetInDb(
+  connector: ConnectorResource,
+  internalId: string,
+  file: microsoftgraph.DriveItem
+) {
+  return MicrosoftNodeResource.upsert({
+    internalId,
+    connectorId: connector.id,
+    lastSeenTs: new Date(),
+    nodeType: "file" as const,
+    name: file.name ?? "",
+    mimeType:
+      "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+    lastUpsertedTs: new Date(),
+  });
+}
+
+async function upsertWorksheetInDb(
+  connector: ConnectorResource,
+  internalId: string,
+  worksheet: microsoftgraph.WorkbookWorksheet
+) {
+  return MicrosoftNodeResource.upsert({
+    internalId,
+    connectorId: connector.id,
+    lastSeenTs: new Date(),
+    nodeType: "worksheet" as const,
+    name: worksheet.name ?? "",
+    mimeType: "text/csv",
+    lastUpsertedTs: new Date(),
+  });
+}
+
+async function upsertTable(
+  connector: ConnectorResource,
+  internalId: string,
+  spreadsheet: microsoftgraph.DriveItem,
+  worksheet: microsoftgraph.WorkbookWorksheet,
+  rows: string[][],
+  loggerArgs: object
+) {
+  const dataSourceConfig = await dataSourceConfigFromConnector(connector);
+
+  const tableName = slugify(
+    `${spreadsheet.name?.substring(0, 16)}-${worksheet.name?.substring(0, 16)}`
+  );
+
+  const tableDescription = `Structured data from the Excel Spreadsheet (${spreadsheet.name}) and sheet (${worksheet.name}`;
+
+  const csv = stringify(rows);
+
+  // Upserting is safe: Core truncates any previous table with the same Id before
+  // the operation. Note: Renaming a sheet in Google Drive retains its original Id.
+  await upsertTableFromCsv({
+    dataSourceConfig,
+    tableId: internalId,
+    tableName,
+    tableDescription,
+    tableCsv: csv,
+    loggerArgs: {
+      connectorId: connector.id,
+      sheetId: internalId,
+      spreadsheetId: spreadsheet.id ?? "",
+    },
+    truncate: true,
+  });
+
+  logger.info(loggerArgs, "[Spreadsheet] Table upserted.");
+}
+
+async function processSheet(
+  client: Client,
+  connector: ConnectorResource,
+  spreadsheet: microsoftgraph.DriveItem,
+  internalId: string,
+  worksheet: microsoftgraph.WorkbookWorksheet
+): Promise<boolean> {
+  if (!worksheet.id) {
+    return false;
+  }
+  logger.info({ internalId }, "Processing sheet");
+  const content = await getWorksheetContent(client, internalId);
+  const loggerArgs = {
+    connectorId: connector.id,
+    worksheetId: internalId,
+    sheet: {
+      id: worksheet.id,
+      name: worksheet.name,
+    },
+  };
+
+  logger.info(
+    { loggerArgs },
+    "[Spreadsheet] Processing sheet in Microsoft Excel."
```

Comment:
As we use the same logs as in Google Spreadsheet, can we add an extra logger arg to reflect the connector type?